### DISCUSSION TITLE
Modify telegraf classes by changing their prototypes

### DISCUSSION THREAD

**Discussion by minasploit** [telegraf/telegraf#1408] [Q&A] [answered]
I wanted to add some fields to existing types. For example, I wanted to add a full_name field to the User prototype that resolves accordingly to a full name or just a first name as such...

`User.prototype.full_name = function () {
        return this.last_name != undefined ? this.first_name + " " + this.last_name : this.first_name;
    }`

But I can't, for the life of me, import the right module to access the User object. Is there any way I can import telegraf classes like User and Chat so their prototypes can be modified?

**Comment by wojpawlik** [comment, marked as answer]
They come from https://github.com/KnorpelSenf/typegram, but they're not classes:

> this package just exposes a huge load of `interface`s [...].
> It contains zero bytes of executable code.

```ts
function fullName(user: User) {
  return user.last_name !== undefined
    ? `${user.first_name} ${user.last_name}`
    : user.first_name;
}
```

**Reply by minasploit** [reply]
That solution doesn't 

1. Attach a new property to the type itself
2. Solve the solution and just adds an unnecessary package which can be done natively by Telegraf itself

**Reply by wojpawlik** [reply]
What "unnecessary package"?

**Reply by minasploit** [reply]
Ohhh, I apologize... I had thought you meant for me to add the https://github.com/KnorpelSenf/typegram package and access the types from there. Sorry.

Other than your proposed solution, which creates a new function rather than extend the current classes, is there not any other opeion?
